### PR TITLE
rpm 4.18 and test fixes

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
@Homebrew/brew this is going to fail the style check on the usage of RPM macros. Any suggestions how to avoid that? ` %<foo>s` doesn't work so well in RPMs

File: Formula/rpm.rb [hunk 1]
```diff
@@ -105,10 +78,30 @@ def install
...
+
+    system bin/"rpmdb", "--initdb", "--root=#{testpath}"
+    system bin/"rpm", "-vv", "-qa", "--root=#{testpath}"
+    assert_predicate testpath/var/"lib/rpm/rpmdb.sqlite", :exist?,
+                     "Failed to create 'rpmdb.sqlite' file"
+
+    %w[SPECS BUILD BUILDROOT].each do |dir|
+      (testpath/"rpmbuild/#{dir}").mkpath
+    end
+
```
Comment (SMillerDev) [inline Formula/rpm.rb:99]:
```suggestion

```

File: Formula/rpm.rb [hunk 2]
```diff
@@ -105,10 +78,29 @@ def install
...
+    safe_system bin/"rpmdb", "--initdb" unless (var/"lib/rpm/rpmdb.sqlite").exist?
   end
 
-  def test_spec
-    <<~EOS
+  test do
+    ENV["HOST"] = "test"
+    (testpath/".rpmmacros").write <<~EOS
+      %_topdir  %(echo $HOME)/rpmbuild
+      %_tmppath	%{_topdir}/tmp
```
Comment (SMillerDev) [inline Formula/rpm.rb:88]:
```suggestion
      %_tmppath	%_topdir/tmp
```

File: Formula/rpm.rb [hunk 3]
```diff
@@ -122,40 +114,33 @@ def test_spec
 
       %prep
       %build
+      echo "hello brew" > test
+
       %install
-      mkdir -p $RPM_BUILD_ROOT/tmp
-      touch $RPM_BUILD_ROOT/tmp/test
+      install -d $RPM_BUILD_ROOT/%{_docdir}
```
Comment (SMillerDev) [inline Formula/rpm.rb:120]:
```suggestion
      install -d $RPM_BUILD_ROOT/%_docdir
```

File: Formula/rpm.rb [hunk 4]
```diff
@@ -122,40 +114,33 @@ def test_spec
...
+      cp test $RPM_BUILD_ROOT/%{_docdir}/test
```
Comment (SMillerDev) [inline Formula/rpm.rb:121]:
```suggestion
      cp test $RPM_BUILD_ROOT/%_docdir/test
```

File: Formula/rpm.rb [hunk 5]
```diff
@@ -122,40 +114,33 @@ def test_spec
...
 
       %files
-      /tmp/test
+      %{_docdir}/test
```
Comment (SMillerDev) [inline Formula/rpm.rb:124]:
```suggestion
      %_docdir/test
```

**Comment by SMillerDev** [comment]
It works on my machine 😅 
```
  checking for mkfifoat... no
  configure: error: function required by rpm
```

**Comment by cho-m** [comment]
https://github.com/rpm-software-management/rpm/issues/2222

Sounds like `mkfifoat` was added in Ventura.

File: Formula/rpm.rb [hunk 6]
```diff
@@ -1,9 +1,8 @@
 class Rpm < Formula
   desc "Standard unix software packaging tool"
   homepage "https://rpm.org/"
-  url "http://ftp.rpm.org/releases/rpm-4.17.x/rpm-4.17.0.tar.bz2"
-  mirror "https://ftp.osuosl.org/pub/rpm/releases/rpm-4.17.x/rpm-4.17.0.tar.bz2"
-  sha256 "2e0d220b24749b17810ed181ac1ed005a56bbb6bc8ac429c21f314068dc65e6a"
+  url "https://ftp.osuosl.org/pub/rpm/releases/rpm-4.18.x/rpm-4.18.0.tar.bz2"
+  sha256 "2a17152d7187ab30edf2c2fb586463bdf6388de7b5837480955659e5e9054554"
   license "GPL-2.0-only"
   revision 1
```
Comment (SMillerDev) [inline Formula/rpm.rb:7]:
```suggestion
```

**Comment by SMillerDev** [comment]
Should probably try the suggestion from https://github.com/rpm-software-management/rpm/issues/2332#issuecomment-1367981310 here too

**Comment by p-linnane** [comment]
Should we apply the flat namespace patch just on macOS?

**Review by p-linnane — APPROVED** [review]

File: Formula/rpm.rb [hunk 7]
```diff
@@ -24,14 +22,11 @@ class Rpm < Formula
...
 
-  # We need autotools for the Lua patch below. Remove when the patch is no longer needed.
-  depends_on "autoconf" => :build
-  depends_on "automake" => :build
-  depends_on "libtool" => :build
   depends_on "gettext"
   depends_on "libarchive"
   depends_on "libmagic"
   depends_on "lua"
+  depends_on macos: :ventura
```
Comment (kamronbatman) [inline Formula/rpm.rb:29]: @SMillerDev, why was this added? In a corporate environment where upgrades are controlled, this will effectively break those individuals since there doesn't seem to be a way to install a specific version.

Comment (kamronbatman) [reply Formula/rpm.rb:29]: I see there is a dependency on `mkfifoat` for rpm itself. Is there a way around this?
